docs/DesignDocs/libc_pthread_proposal.md

```diff
@@ -0,0 +1,80 @@
+# To support pthread in Open Enclave SDK
+
+Currently, pthread management APIs are not implemented in OE SDK while they are
+implemented in SGX SDK. The pthread APIs are added when porting MKL-DNN to SGX SDK.
+To get better performance, MKL-DNN should be configured to build with OpenMP which
+depends on pthread.
+
+## New APIs
+
+### oe_thread_create
+The function is used to create a new thread inside enclave.
+
+oe_thread_create is an internal function which will be wrapped to an external API pthread_create.
+
+The work sequence should be like,
+* Parent thread(Enclave): call pthread_create() inside the enclave codes written by users.
+* Parent thread(Enclave): initialize some structures and execute OCALL - OE_OCALL_THREAD_CREATE.
+* Parent thread(Host): allocate some resources and call oe_thread_create() to create a new thread.
+* Child thread(Host): execute ECALL - OE_ECALL_THREAD_CREATE_ROUTINE if new thread is created successfully.
+* Child thread(Enclave): get the user defined hook function and execute it.
+
+**NOTE**: Currently the parameter of attributes for the created thread is unused. So the created threads are
```

Comment:
The parameter should not be "unused".  You should explicitly check that it contains the behavior you support (e.g., == NULL) and fail with EINVAL if not.  Otherwise future additions can have app compact problems.

Reply:
Seems the attr parameter of locks initialize function is "unused".  Should these pthread functions be aligned?